src/main/java/io/cdap/plugin/gcp/bigquery/sink/AvroRecordWriter.java

```diff
@@ -58,6 +59,7 @@ public AvroRecordWriter(Schema writerSchema, GenericData dataModel, CodecFactory
     this.compressionCodec = compressionCodec;
     this.outputStream = outputStream;
     this.syncInterval = syncInterval;
+    createFileWriter(Schema.create(Schema.Type.NULL), ByteStreams.nullOutputStream());
```

Comment:
Hi there! If I may drop a suggestion/question:
Why not setting writerSchema here as first input and outputStream as second input and stop checking for schema change for each record in "write"? Anyway, even if it fails here because of schema issues, it's only 1 step earlier than at the BQ Load step.
At least in my specific use case, I have a "large" schema and all incoming records contain fields that are a subset of that "large" schema. Thus if we set the large schema here (coming from the output schema that I set in the UI), `mAvroFileWriter` will never need to be re-created whereas if we keep getting schemas from the records, it will have to be sync/close/recreated every X records (though unsure if that has significant performance impact, particularly if the default is 'only' 8MB/file). Correct me if I'm wrong as I'm not a CDAP expert.

But I understand that this was only intended to initialise prevSchema/mAvroFileWriter and if you prefer not to introduce additional complexity in this PR, it also makes sense to me.

PS: You never close ByteStreams.nullOutputStream() later, so even though it will never grow more than the object initialization, can it lead to some problem or is it just fine because de-referencing it will auto-close it?

Reply:
- When the AvroRecordWriter is initialized in the constructor the value of `writerSchema` is `null`. the `writerSchema` is rather pulled from every record. Hence cannot initialize `mAvroFileWriter` with null object.

- The current is quite generic to handle different schemas . Even in your usecase, the `mAvroFileWriter.sync/close` will happen only once, since it's reloaded only if there is a new schema. 

- `ByteStreams.nullOutputStream()` is passed on to `mAvroFileWriter` , and when we call `mAvroFileWriter.close()`, it internally closes the outstream as well.